In [1]:
import selenium
import pandas as pd
from selenium import webdriver
import warnings
warnings.filterwarnings('ignore')
from selenium.common.exceptions import StaleElementReferenceException, NoSuchElementException
from selenium.webdriver.common.by import By

import time

In [2]:
driver= webdriver.Chrome(r'C:\Users\user\Downloads\chromedriver_win32 (2)\chromedriver.exe')
time.sleep(2)

In [3]:
url='https://www.cardekho.com/used-cars+in+new-delhi'
driver.get(url)

In [4]:
car_name=[]
car_variant=[]
car_price=[]
car_distance=[]
car_fuel=[]
car_work=[]
car_customer_care=[]
time.sleep(2)

In [5]:
time.sleep(4)
car_n=driver.find_elements(By.XPATH,"//div[@class='gsc_col-xs-7  carsName']")
for i in car_n[0:5000]:
    car_name.append(i.text.split('\n')[0]) 

    
car_v=driver.find_elements(By.XPATH,"//div[@class='gsc_col-xs-7  carsName']")
for j in car_v[0:5000]:
    car_variant.append(j.text.split('\n')[1])

car_p=driver.find_elements(By.XPATH,"//div[@class='gsc_col-xs-7  carsName']")
for k in car_p[0:5000]:
    car_price.append(k.text.split('\n')[2])

    
car_d=driver.find_elements(By.XPATH,"//div[@class='gsc_col-xs-7  carsName']")
for l in car_d[0:5000]:
    car_distance.append(l.text.split('\n')[3])

    
car_f=driver.find_elements(By.XPATH,"//div[@class='gsc_col-xs-7  carsName']")
for m in car_f[0:5000]:
    car_fuel.append(m.text.split('\n')[4])

car_w=driver.find_elements(By.XPATH,"//div[@class='gsc_col-xs-7  carsName']")
for n in car_w[0:5000]:
    car_work.append(n.text.split('\n')[5])

    
car_cc=driver.find_elements(By.XPATH,"//div[@class='gsc_col-xs-7  carsName']")
for o in car_cc[0:5000]:
    car_customer_care.append(o.text.split('\n')[6])


In [6]:
len(car_name),len(car_variant),len(car_price),len(car_distance),len(car_fuel),len(car_work),len(car_customer_care)

(3394, 3394, 3394, 3394, 3394, 3394, 3394)

Note: it is showing 3394 results instead of 5000 because of its website at last  it says that is all folks

In [7]:
df=pd.DataFrame({'BRAND':car_name,'VARIANT':car_variant,'PRICE':car_price,'DISTANCE':car_distance,'FUEL':car_fuel,'How it WORK':car_work,'CUSTOMER CARE':car_customer_care})
df

BRAND           VARIANT  \
0      2018 Maruti Alto 800               LXI   
1      2016 Mahindra XUV500    W10 1.99 mHawk   
2          2019 Maruti Ciaz        Alpha BSIV   
3         2021 Nissan Kicks            1.5 XV   
4          2019 Maruti Ciaz        Delta BSIV   
...                     ...               ...   
3389  2017 Mahindra Scorpio      1.99 S4 Plus   
3390   2020 Maruti S-Presso               VXI   
3391    2017 Maruti Wagon R           CNG LXI   
3392      2018 Renault KWID  1.0 RXT Optional   
3393     2018 Hyundai Creta        1.4 E Plus   

                                    PRICE      DISTANCE    FUEL How it WORK  \
0                              ₹3.15 Lakh    44,711 kms  Petrol      Manual   
1     ₹9.10 Lakh₹ 10,68,500Save ₹1,58,500  1,08,939 kms  Diesel      Manual   
2         ₹9.34 Lakh₹ 9,40,000Save ₹6,000    33,206 kms  Petrol      Manual   
3     ₹9.52 Lakh₹ 10,59,000Save ₹1,07,000     9,799 kms  Petrol      Manual   
4         ₹8.17 Lakh₹ 8,26,000Save ₹9,000    34,474 kms  Petrol      Manual   
...                                   ...           ...     ...         ...   
3389                           ₹8.00 Lakh    10,000 kms  Diesel      Manual   
3390                           ₹5.00 Lakh    10,000 kms  Petrol      Manual   
3391                           ₹3.74 Lakh    75,000 kms     CNG      Manual   
3392                           ₹3.26 Lakh    24,000 kms  Petrol      Manual   
3393                           ₹9.50 Lakh    68,000 kms  Diesel      Manual   

     CUSTOMER CARE  
0        Trustmark  
1        Trustmark  
2        Trustmark  
3        Trustmark  
4        Trustmark  
...            ...  
3389       Partner  
3390       Partner  
3391       Partner  
3392       Partner  
3393       Partner  

[3394 rows x 7 columns]

In [9]:
df.to_csv(r'C:\Users\user\Desktop\FRTech Internship Project\carprediction\car prediction.csv')

In [10]:
driver.close()

# 

#             THANK YOU